src/Models/Environment.php

```diff
@@ -996,27 +996,55 @@ public function sftpConnectionInfo()
     /**
      * "Wake" a site
      *
+     * @param int $maxRetries Maximum number of retries
+     * @param int $delay Delay between retries in seconds
      * @return array
+     *
+     * @throws \Pantheon\Terminus\Exceptions\TerminusException
      */
-    public function wake(): array
+    public function wake(int $maxRetries = 3, int $delay = 5): array
     {
-        $domains = array_filter(
-            $this->getDomains()->all(),
-            function ($domain) {
-                $domain_type = $domain->get('type');
-                return (!empty($domain_type) && "platform" == $domain_type);
+        $attempt = 0;
+        do {
+            $attempt++;
+            try {
+                $domains = array_filter(
+                    $this->getDomains()->all(),
```

Comment:
Do we really need to do this in the retries loop? Isn't this function going through the regular collections/models stuff that already retries? (I'm worried that this will attempt to hit the platform way more than expected)